substrate/runtime/contract/src/vm/mod.rs

```diff
@@ -0,0 +1,399 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate. If not, see <http://www.gnu.org/licenses/>.
+
+//! This module provides a means for executing contracts
+//! represented in wasm.
+
+use exec::{CallReceipt, CreateReceipt};
```

Comment:
this code mostly preserved from [`src/vm.rs`](https://github.com/paritytech/substrate/blob/690f52f5ab269b6c35aba9e7056f3362291c7f5a/substrate/runtime/contract/src/vm.rs)